### ISSUE TITLE
PDOException on GCP: remaining connection slots are reserved for non-replication superuser connections

### ISSUE THREAD

**Issue by norbert-yoimo** [dmaicher/doctrine-test-bundle#50]
The bundle works great locally (the time it takes for our test suite to run went from 2:30 to 0:30), however some tests fail (namely, 3 out of 99) with the above error when we try to run our tests in Jenkins, which is deployed on Google cloud, and uses Google cloud sql (postgres).

Do you have any pointers on how to resolve this, or what might be the cause?

The full error is:
```

1) Tests\xxxBundle\Payment\PaymentServiceTest::testFindNextYearlyPaymentDate with data set #0 ('2017-01-01', '2018-01-01')
Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[08006] [7] FATAL:  remaining connection slots are reserved for non-replication superuser connections

/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractPostgreSQLDriver.php:85
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:145
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php:66
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
/-/vendor/dama/doctrine-test-bundle/src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php:42
/-/var/cache/test/appTestDebugProjectContainer.php:1208
/-/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:336
/-/var/cache/test/appTestDebugProjectContainer.php:1258
/-/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:336
/-/tests/xxxBundle/xxxTestCase.php:20

Caused by
Doctrine\DBAL\Driver\PDOException: SQLSTATE[08006] [7] FATAL:  remaining connection slots are reserved for non-replication superuser connections

/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:47
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php:45
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
/-/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
/-/vendor/dama/doctrine-test-bundle/src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php:42
/-/var/cache/test/appTestDebugProjectContainer.php:1208
/-/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:336
/-/var/cache/test/appTestDebugProjectContainer.php:1258
/-/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:336
/-/tests/xxxBundle/xxxTestCase.php:20

```

**Comment by dmaicher** [comment]
Hmm I'm not so familiar with the Google Clould SQL environment :confused: 

How exactly are you running your test suite? Is it one php process? It seems for some reason too many connections are created and not closed properly.

**Comment by norbert-yoimo** [comment]
It's one php process, invoked simply by `php ./vendor/bin/phpunit`.

Does the bundle do anything like opening multiple persistent connections?

**Comment by dmaicher** [comment]
This bundle keeps connections statically in the php process:

https://github.com/dmaicher/doctrine-test-bundle/blob/master/src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticDriver.php#L48

Maybe for some reason this key changes in your case? And too many connections are created?

https://github.com/dmaicher/doctrine-test-bundle/blob/master/src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticDriver.php#L46

**Comment by dmaicher** [comment]
@norbert-yoimo any chance to debug this a bit yet?

**Comment by norbert-yoimo** [comment]
Sorry not yet, but it's on my todo list

**Comment by norbert-yoimo** [comment]
I had an error in the CI pipeline. Previously it was the test harness that was responsible for generating the database, but after installing this bundle, I removed this logic, so that the database does not get dropped/recreated between tests.

This caused all the tests to try to connect to a non-existant database. I have moved the database creation to be part of the CI pipeline, but now I'm running into a different problem.
